app/controllers/products_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  # GET /products
+  # GET /products.json
+  def index
+    @products = Product.all
+  end
+
+  # GET /products/1
+  # GET /products/1.json
+  def show
+  end
+
+  # GET /products/new
+  def new
+    @product = Product.new
+  end
+
+  # GET /products/1/edit
+  def edit
+  end
+
+  # POST /products
+  # POST /products.json
+  def create
+    @product = Product.new(product_params)
+
+    respond_to do |format|
+      if @product.save
+        format.html { redirect_to @product, notice: 'Product was successfully created.' }
+        format.json { render :show, status: :created, location: @product }
```

Comment:
do you use the json format anywhere? if not, just remove it. we don't want code we don't use

Reply:
removed ✓